walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -113,12 +115,51 @@
 
     // optional attendees
     // repeat the same process as the manatory attendees to narrow time options to fit optional attendees
+    
+    times = removeConflictingTimes(times, optionalAttendees, events, request.getDuration(), false);
 
-    times = removeConflictingTimes(times, optionalAttendees, events, request.getDuration());
-
-    // if optional attendees are unavailable, return times for mandatory attendees
+    // If no time exists for all optional and mandatory attendees, find the time slot(s) 
+    // that allow mandatory attendees and the greatest possible number of optional attendees to attend.
     if(times.isEmpty() && !(attendees.isEmpty())){
+
         times = timesBeforeOptional;
+
+        // breaks times into smaller peices that are available for different optional attendees
+        times = removeConflictingTimes(times, optionalAttendees, events, request.getDuration(), true);
+
+        int lowestConflicts = optionalAttendees.size() - 1;
+        ArrayList<TimeRange> mostPeopleAvailable = new ArrayList<>();
```

Comment:
this variable name and its type don't seem to match. maybe something like bestTimeRanges

Reply:
Alright, cool I changed that.